### PR TITLE
Fix lint CI and split it into multiple jobs

### PR DESCRIPTION
**Pull request by andresilva91**
## Motivation

On #1737 we added `continue-on-error` on the job level. AFAIU that will succeed even if the entire jobs fails with a cancellation for example, which will happen when we timeout.
This was causing this job to silently succeed while timing out for a while.

## Proposal

Split the different lints into their own jobs, adding parallelism and also giving us individual failures

## Test Plan

CI

## Release Plan

- Nothing to do / These changes follow the usual release cycle.

### CODE REVIEW

**Comment by andresilva91** [comment]
* **#2785** <a href="https://app.graphite.dev/github/pr/linera-io/linera-protocol/2785?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @andresilva91 and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

**Comment by ma2bd** [comment]
Should we just not use `continue-on-error` and split the linting jobs in several parallel workflows?

**Comment by andresilva91** [comment]
We could! That'll be a lot of jobs though 😅 And the only time consuming one here is the `check-all-features` which I'm splitting off to a separate job in the following PR. Happy to split it further though

**Comment by andresilva91** [comment]
Looking at the documentation more, I think that if the last job without `continue-on-error` succeeds, the job will succeed regardless of the errors from the other jobs. So I agree, it seems the best solution here is to split everything in multiple parallel lint jobs. I'll do that

File: .github/workflows/rust.yml [hunk 1]
```diff
@@ -242,17 +353,76 @@ jobs:
...
+      run: |
+        cargo install --git https://github.com/ma2bd/cargo-all-features --branch workspace_metadata --locked
+    - name: Install Protoc
+      uses: arduino/setup-protoc@v1
+      with:
+        repo-token: ${{ secrets.GITHUB_TOKEN }}
+    - name: Run cargo check-all-features
+      run: |
+        cargo check-all-features
+        cargo check-all-features --all-targets
```
Comment (andresilva91) [inline .github/workflows/rust.yml:428]: Also, isn't it redundant to run both of these? I'm removing `cargo check-all-features`

File: .github/workflows/rust.yml [hunk 2]
```diff
@@ -160,10 +160,9 @@ jobs:
         cd linera-views
         WASM_BINDGEN_TEST_TIMEOUT=300 wasm-pack test --chrome --headless -- --features web-default
 
-  lint:
+  unexpected-chain-load-operations:
```
Comment (ma2bd) [inline .github/workflows/rust.yml:163]: Should we keep `lint` somewhere in the name? Maybe `lint-unexpected-chain-load-operations`

Comment (ma2bd) [reply .github/workflows/rust.yml:163]: (even better) Shouldn't we group the fast lints together to save git-checkout time?

Comment (ma2bd) [reply .github/workflows/rust.yml:163]: Never mind "checkout" is surprisingly fast.

"Clear some space" is slow, though. In fact, we probably shouldn't do it for linters.

File: .github/workflows/rust.yml [hunk 3]
```diff
@@ -176,6 +175,19 @@ jobs:
     - name: Check for unexpected chain load operations
       run: |
         ./scripts/check_chain_loads.sh
+
+  check-copyright-headers:
```
Comment (ma2bd) [inline .github/workflows/rust.yml:179]: nit: Maybe `lint-check-copyright-headers`. In fact `lint-cargo-clippy` is probably logical after that

File: .github/workflows/rust.yml [hunk 4]
```diff
@@ -227,12 +296,54 @@ jobs:
                 cargo rdme --check --no-fail-on-warnings -w "${dir_name}"
             fi
         done)
+
+  wasm-application-lints:
```
Comment (ma2bd) [inline .github/workflows/rust.yml:300]: `lint-wasm-application`?

File: linera-client/src/unit_tests/mod.rs [hunk 5]
```diff
@@ -3,4 +3,5 @@
 
 mod chain_listener;
 mod util;
+#[cfg(feature = "fs")]
```
Comment (ma2bd) [inline linera-client/src/unit_tests/mod.rs:6]: That wasn't so bad :)

Comment (andresilva91) [reply linera-client/src/unit_tests/mod.rs:6]: Yeah 🤣

File: .github/workflows/rust.yml (hunk 1 above)
Comment (ma2bd) [reply .github/workflows/rust.yml]: I don't remember it but I must have had a reason

**Review by ma2bd — APPROVED** [review]

**Comment by andresilva91** [comment]
### Merge activity

* **Nov 5, 10:19 AM EST**: A user started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/linera-io/linera-protocol/2785).

**Pull request merged**